john_implementation/src/john_implementation/core.clj

```diff
@@ -0,0 +1,24 @@
+(ns john-implementation.core
+  (:require [org.httpkit.client :as http-kit]
+            [cheshire.core :refer :all])
+  (:gen-class))
+
+
+(defn get-submissions
+  [user_id]
+  (def response  (http-kit/get (str "https://hacker-news.firebaseio.com/v0/user/" user_id ".json")))
+    (get (parse-string (:body @response)) "submitted")
+)
+
+
+; This takes a userid, finds all submissions, and returns a list of post id's
+(defn sub-to-list
+  [user_id]
+  (def list (get-submissions user_id)))
```

Comment:
Shouldn't change state (by `def list`) inside a function. Return `(get-submissions user-id)` and use that return value. `(println (sub-to-list "elemje"))`

Reply:
This might be a dumb question, but why not change state? I might be misunderstanding Clojure, but doesn't it evaluate the variable declaration after the constant value of submissions is returned by `get-submissions`

Reply:
I thought the whole purpose of functional programming is to reduce side effects. This is equivalent of declaring a global variable and modifying it and it is hard to tell what `list` is at any point. An example would be if you define a function that needs to maps that list

```
(defn test [] (map (fn [x] (+ x 1), list))
```
you are only allowed to call the function once `sub-to-list` is called or you `list` will not be defined.

Or I could be wrong.